lsp-protocol.el

```diff
@@ -136,9 +136,10 @@ Example usage with `dash`.
                                                        (s-dashed-words (symbol-name interface))
                                                        (substring (symbol-name label) 1)))
                                    (object value)
-                                 ,(if lsp-use-plists
-                                      `(plist-put object ,name value)
-                                    `(puthash ,(lsp-keyword->string name) value object)))))
+                                 ,@(if lsp-use-plists
+                                       `((plist-put object ,name value))
```

Comment:
This operation seems redundant, can you elaborate?

Reply:
The document for `plist-put` suggest it. It will work when `object` is `nil`.
In case of `hash-table`, it will throw error when `object` is `nil`, but for `plist` it will innocently pass, seems to return correct value but actually not changing `object`.

Reply:
This generates setter, so the value is returned to the caller of the function

Reply:
Sorry I missed this comment. Exactly because it's setter, we should expect it to set the value into object.
The returns are actually irrelevant here, since setter, as its name, is mostly used for side effect only.
However, since we're passing the object via parameter directly, I agree that `setq` is not useful here.